hazelcast/src/main/java/com/hazelcast/ringbuffer/Ringbuffer.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.ringbuffer;
+
+import com.hazelcast.core.DistributedObject;
+import com.hazelcast.core.ICompletableFuture;
+import com.hazelcast.core.IFunction;
+import com.hazelcast.spi.annotation.Beta;
+
+import java.util.Collection;
+
+/**
+ * A Ringbuffer is a data-structure where the content is stored in a ring like structure. A ringbuffer has a capacity so it
+ * won't grow beyond that capacity and endanger the stability of the system. If that capacity is exceeded, than the oldest
+ * item in the ringbuffer is overwritten.
+ *
+ * The ringbuffer has 2 always incrementing sequences:
+ * <ol>
+ * <li>
+ * tailSequence: this is the side where the youngest item is found. So the tail is the side of the ringbuffer where
+ * items are added to.
+ * </li>
+ * <li>
+ * headSequence: this is the side where the oldest items are found. So the head is the side where items gets
+ * discarded.
+ * </li>
+ * </ol>
+ * The items in the ringbuffer can be found by a sequence that is in between (inclusive) the head and tail sequence.
+ *
+ * If data is read from a ringbuffer with a sequence that is smaller than the headSequence, it means that the data
+ * is not available anymore and a {@link StaleSequenceException} is thrown.
+ *
```

Comment:
Anyway, the correct way is to treat `<p></p>` as a paragraph *container*, not paragraph *break*. So `<p>The paragraph's contents</p>`.